### PR TITLE
fix(windows): fix win32 errors

### PR DESCRIPTION
**Pull request by dylan-conway**
### What does this PR do?
fixes some test regressions

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### CODE REVIEW

File: src/windows.zig [hunk 1]
```diff
@@ -2999,6 +2999,7 @@ pub fn translateWinErrorToErrno(err: win32.Win32Error) bun.C.E {
         .NOT_ENOUGH_MEMORY => .NOMEM,
         .OUTOFMEMORY => .NOMEM,
         .INVALID_PARAMETER => .INVAL,
+        .ALREADY_EXISTS => .EXIST,
```
Comment (Jarred-Sumner) [inline src/windows.zig]: @paperdave why did we do this? we have https://github.com/oven-sh/bun/blob/58132dc9dace934f7a71be46a920d29c4e10cf9f/src/windows_c.zig#L706-L805

Comment (Jarred-Sumner) [reply src/windows.zig]: windows errno.h:
```c
#define EPERM           1
#define ENOENT          2
#define ESRCH           3
#define EINTR           4
#define EIO             5
#define ENXIO           6
#define E2BIG           7
#define ENOEXEC         8
#define EBADF           9
#define ECHILD          10
#define EAGAIN          11
#define ENOMEM          12
#define EACCES          13
#define EFAULT          14
#define EBUSY           16
#define EEXIST          17
#define EXDEV           18
#define ENODEV          19
#define ENOTDIR         20
#define EISDIR          21
#define ENFILE          23
#define EMFILE          24
#define ENOTTY          25
#define EFBIG           27
#define ENOSPC          28
#define ESPIPE          29
#define EROFS           30
#define EMLINK          31
#define EPIPE           32
#define EDOM            33
#define EDEADLK         36
#define ENAMETOOLONG    38
#define ENOLCK          39
#define ENOSYS          40
#define ENOTEMPTY       41
```

Comment (paperclover) [reply src/windows.zig]: > why did we do this?

i simply did not find/realize that list was there.

Comment (paperclover) [reply src/windows.zig]: i think the translate ntstatus one is still needed, uncertain if the codes align.

very cautious about incorrect enum casts

**Comment by Jarred-Sumner** [comment]
I think the assertion for unknown has to be changed to a debugWarn

**Pull request merged**